tests/appservice/test_api.py

```diff
@@ -74,11 +75,86 @@ async def get_json(
             args: Mapping[Any, Any],
             headers: Mapping[Union[str, bytes], Sequence[Union[str, bytes]]],
         ) -> List[JsonDict]:
-            # Ensure the access token is passed as both a header and query arg.
-            if not headers.get("Authorization") or not args.get(b"access_token"):
+            # Ensure the access token is passed as a header.
+            if not headers or not headers.get("Authorization"):
                 raise RuntimeError("Access token not provided")
+            # ... and not as a query param
+            if args and args.get(b"access_token"):
```

Comment:
We can remove the first bit if this is a mapping and we only care if the param exists at all (i.e. if it is blank that's still an error):

```suggestion
            if b"access_token" in args:
```